### PR TITLE
(RE-16550): Debian 12 bookworm FOSS build target

### PR DESCRIPTION
**Pull request by h0tw1r3**

### CODE REVIEW

File: resources/puppetlabs/lein-ezbake/template/foss/ext/build_defaults.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 default_cow: 'base-bionic-i386.cow'
-cows: 'base-bionic-i386.cow base-buster-i386.cow base-focal-i386.cow base-bullseye-i386.cow base-jammy-i386.cow'
+cows: 'base-bionic-i386.cow base-buster-i386.cow base-focal-i386.cow base-bullseye-i386.cow base-jammy-i386.cow base-bookworm-i386.cow'
```
Comment (smortex) [inline resources/puppetlabs/lein-ezbake/template/foss/ext/build_defaults.yaml:3]: Is this required?  As far as I can see, most packages are architecture agnostic with the exception of the `pdk`, `puppet-agent` and `puppet-bolt` ones, and for these no i386 packages are available for the oldstable release (both for Puppet 7 and Puppet 8) so I don't expect you will build some for the stable release?

https://apt.puppet.com/pool/bullseye/puppet7/p/pdk/index.html
https://apt.puppet.com/pool/bullseye/puppet7/p/puppet-agent/index.html
https://apt.puppet.com/pool/bullseye/puppet7/p/puppet-bolt/index.html
https://apt.puppet.com/pool/bullseye/puppet8/p/pdk/index.html
https://apt.puppet.com/pool/bullseye/puppet8/p/puppet-agent/index.html
https://apt.puppet.com/pool/bullseye/puppet8/p/puppet-bolt/index.html

Comment (h0tw1r3) [reply resources/puppetlabs/lein-ezbake/template/foss/ext/build_defaults.yaml:3]: Legacy stuff. It should be cleaned up at some point. build.rake parses the cows, and likely a bunch of puppet internal pipelines pass it around.

Comment (smortex) [reply resources/puppetlabs/lein-ezbake/template/foss/ext/build_defaults.yaml:3]: Fine, thanks for the clarification!

**Review by smortex — APPROVED** [review]
/me wants Debian 12 packages for puppetserver and puppetdb :sob: 

Approving, even if my approval will probably not help at all :cry:

**Review by tuxmea — APPROVED** [review]

**Comment by kenyon** [comment]
Typo in PR title: Debain -> Debian

**Comment by joshcooper** [comment]
@h0tw1r3 could you rebase and resolve conflicts?

**Comment by span786** [comment]
@puppetlabs/release-engineering / @puppetlabs/dumpling / @puppetlabs/skeletor 
I successfully generated the lein-ezbake JAR using the changes from this PR: [lein-ezbake-2.6.2.jar.zip](https://github.com/user-attachments/files/16728969/lein-ezbake-2.6.2.jar.zip). However, there isn't an ad-hoc Jenkins pipeline available to test if the JAR is functioning correctly. Can we merge this PR and test it in the main pipeline [here](https://jenkins-enterprise.delivery.puppetlabs.net/view/all/job/enterprise_ezbake_release-clj_main/)?

**Comment by bastelfreak** [comment]
All our lives would be easier if the pipelines would be public again, at least for the Vox Pupuli PMC and Puppet Partners.

**Comment by justinstoller** [comment]
I believe the practice before was to build a puppetserver and/or puppetdb package using the changes in a PR and then ensure they install on the target system. Once that is manually tested, ideally by both the PR author and reviewer, then the PR can be merged and a new ezbake released by the pipeline you linked, @span786 .

And Tim is also correct.

**Review by justinstoller — APPROVED** [review]

**Comment by justinstoller** [comment]
Here is a lightly edited log of me validating this PR as an example: [ezbake.log](https://github.com/user-attachments/files/16731492/ezbake.log)

**Pull request merged**